osmo_jupyter/dataset.py

```diff
@@ -0,0 +1,267 @@
+        ],
+        sort=True,
+    )
+
+    calibration_data = pd.concat(
+        [
+            open_calibration_log_file(calibration_log_filepath)
+            for calibration_log_filepath in calibration_log_filepaths
+        ],
+        sort=True,
+    )
+
+    return calibration_data.join(  # By default join happens on index values
+        picolog_data, how="inner"  # Drop any rows without a match in both DataFrames
+    )
+
+
+def get_equilibration_boundaries(equilibration_status: pd.Series) -> pd.DataFrame:
+    """
+        Parse a list of timestamped equilibration statuses into a DataFrame of start
+        and end times of equilibrated states. Assumes all time between unequilibrated
+        and equilibrated input points are unequilibrated.
+
+        Args:
+            equilibration_status: A Series of equilibration status strings.
+        Returns:
+            DataFrame of start and end times of equilibrated states.
+    """
+    # Get a boolean series with timestamped index of equilbrated data points
+    equilibrated_mask = equilibration_status == "equilibrated"
+    leading_edges = equilibrated_mask[equilibrated_mask.astype(int).diff() == 1].index
+    trailing_edges = equilibrated_mask[
+        equilibrated_mask.astype(int).diff(periods=-1) == 1
+    ].index
+
+    # Trim edges to align
+    if len(leading_edges) > len(trailing_edges):
+        leading_edges = leading_edges[:-1]
+    if len(trailing_edges) > len(leading_edges):
+        trailing_edges = trailing_edges[1:]
+
+    return pd.DataFrame({"start_time": leading_edges, "end_time": trailing_edges})
+
+
+def pivot_process_experiment_results_on_ROI(
+    experiment_df: pd.DataFrame,
+    ROI_names: List[str] = None,
+    msorm_types: List[str] = ["r_msorm", "g_msorm", "b_msorm"],
+) -> pd.DataFrame:
+    """
+        Flatten a DataFrame of process experiment results down to one row per image.
+
+        Args:
+            experiment_df: A DataFrame of process experiment summary statistics.
+            ROI_names: Optional. A list of ROI names to select. Defaults to all ROIs present.
+            msorm_types: Optional. A list of MSORM column names to select. Defaults to all RGB channel MSORMs.
+        Returns:
+            DataFrame with one row per image, and msorm values for a subset of ROIs.
+    """
+    if ROI_names:
+        experiment_df = experiment_df[experiment_df["ROI"].isin(ROI_names)]
+
+    # Pivot creates a heirachical multiindex data frame with an index for each 'values' column
+    pivot = experiment_df.pivot(
+        index="timestamp", columns="ROI", values=msorm_types + ["image"]
+    )
+
+    # There's one copy of the image name for each ROI due to the pivot, so pull out the top level index
+    images = pivot["image"][experiment_df["ROI"].values[0]]
+    pivot.drop("image", axis=1, inplace=True)
+    # Flatten the pivot table index, and add images back in
+    pivot.columns = [" ".join(col[::-1]).strip() for col in pivot.columns.values]
```

Comment:
minor: logic here is a little surprising. not sure if black will let you sneak a comment in there but it'd be nice
```suggestion
    pivot.columns = [
       # ("r_msorm", "some ROI name") -> "some ROI name r_msorm"
        " ".join(col[::-1]).strip()
        for col in pivot.columns.values
    ]
```